libnmstate/nm/active_connection.py

```diff
@@ -168,7 +168,10 @@ def nm_active_connection(self):
 
     @property
     def devname(self):
-        return self._nmdev.get_iface()
+        if self._nmdev:
+            return self._nmdev.get_iface()
+        else:
+            return None
```

Comment:
Is this fine? If `_nmdev` is None the logs are going to show something like: `Connection deactivation failed on None: error=unknown`. I am not sure if it will help or confuse the user but I don't have any other solution on mind.

Reply:
I agree with @ffmancera on this.

The change itself may be fine, but the reasoning for it (in the commit message) is not IMO.
If `devname` has been queried, it means it was expected to exist, otherwise it would not have been called.
And the reasoning about OVS service is just masking the real problem (or other future problems).

We should probably blow up much earlier or not reach this situation at all.
For the OVS issue itself, I think we should just not proceed if it is not running, like the NM ovs plugin.

Reply:
> Is this fine? If `_nmdev` is None the logs are going to show something like: `Connection deactivation failed on None: error=unknown`. I am not sure if it will help or confuse the user but I don't have any other solution on mind.

The problem about the missing openvswitch service is tracked at #837 if it is fixed, this code would not be reached.

The error message could also be improved. I guess it should mention the connection name instead of only the device name to reduce ambiguity.

> The change itself may be fine, but the reasoning for it (in the commit message) is not IMO.
> If `devname` has been queried, it means it was expected to exist, otherwise it would not have been called.
> And the reasoning about OVS service is just masking the real problem (or other future problems).

The ActiveConnection does not guarantee that the device exists. And it also does not seem to make sense for virtual devices that NM tries to create. If the activation fails, it does not make sense that there would be a device IMHO.
 
> We should probably blow up much earlier or not reach this situation at all.

If you still think that ActiveConnection should be changed so that `_nmdev` is always present, please go ahead and close this PR.

> For the OVS issue itself, I think we should just not proceed if it is not running, like the NM ovs plugin.

This is tracked at #837

Reply:
> Is this fine? If `_nmdev` is None the logs are going to show something like: `Connection deactivation failed on None: error=unknown`. I am not sure if it will help or confuse the user but I don't have any other solution on mind.

When I wanted to file a bug to track this I noticed there is an easy fix for this: #874